projects/ad2s1210_iio/app/app_config.c

```diff
@@ -0,0 +1,166 @@
+#include "no_os_pwm.h"
+#include "no_os_delay.h"
+
+/******************************************************************************/
+/************************ Macros/Constants ************************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/******************** Variables and User Defined Data Types *******************/
+/******************************************************************************/
+
+/* Trigger GPIO init parameters */
+struct no_os_gpio_init_param trigger_gpio_param = {
+	.port = TRIGGER_GPIO_PORT,
+	.number = TRIGGER_GPIO_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &trigger_gpio_ops,
+	.extra = &trigger_gpio_extra_init_params
+};
+
+/* Trigger GPIO IRQ parameters */
+struct no_os_irq_init_param trigger_gpio_irq_params = {
+	.irq_ctrl_id = 0,
+	.platform_ops = &trigger_gpio_irq_ops,
+	.extra = &trigger_gpio_irq_extra_params
+};
+
+/* PWM init parameters */
+static struct no_os_pwm_init_param pwm_init_params = {
+	.id = 0,
+	.period_ns = CONV_TRIGGER_PERIOD_NSEC,
+	.duty_cycle_ns = CONV_TRIGGER_DUTY_CYCLE_NSEC,
+	.extra = &pwm_extra_init_params,
+	.platform_ops = &pwm_ops
+};
+/* UART init parameters */
+static struct no_os_uart_init_param uart_init_params = {
+	.device_id = 0,
+	.baud_rate = IIO_UART_BAUD_RATE,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.platform_ops = &uart_ops,
+	.extra = &uart_extra_init_params
+};
+
+/* UART descriptor */
+struct no_os_uart_desc *uart_desc;
+struct no_os_gpio_desc *trigger_gpio_desc;
+struct no_os_irq_ctrl_desc *trigger_irq_desc;
+struct no_os_pwm_desc *pwm_desc;
+
+/******************************************************************************/
+/************************** Functions Declarations ****************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************** Functions Definitions *****************************/
+/******************************************************************************/
+
+/**
+ * @brief	Initialize the UART peripheral
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t init_uart(void)
+{
+	return no_os_uart_init(&uart_desc, &uart_init_params);
+}
+
+/**
+ * @brief	Initialize the trigger GPIO and associated IRQ event
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t gpio_trigger_Init(void)
+{
+	int32_t ret;
+
+	/* Configure GPIO as input */
+	ret = no_os_gpio_get(&trigger_gpio_desc, &trigger_gpio_param);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_direction_input(trigger_gpio_desc);
+	if (ret)
+		return ret;
+
+	/* Initialize the IRQ controller */
+	ret = no_os_irq_ctrl_init(&trigger_irq_desc, &trigger_gpio_irq_params);
+	if (ret)
```

Comment:
Brackets can be added around if {}

Reply:
this is a single line return statement around the if? why add brackets? if its mandatory for coding style in this repo, should i add brackets to all the other single line after if statements?

Reply:
Yes, the general coding style for this repository.

Reply:
ok.. i will have to add that everywhere..